Dev - Allow to filter wc_help_tip output

jgreys opened the pull request:
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

This filter hook allows developers to modify the HTML output of the help tip displayed on WooCommerce setting pages.

Closes #36543.

### How to test the changes in this Pull Request:

- [x] Have you followed the [Writing high-quality testing instructions guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions)?

1. In theme's functions.php file, add a callback function to the wc_help_tip filter hook that modifies the help tip HTML. For example:

```php
add_filter( 'wc_help_tip', function($tip_html, $sanitized_tip, $tip, $allow_html) {
	return '<div class="woocommerce-help-tip" data-tip="' . strtoupper($sanitized_tip) . '"></div>';
}, 10, 4 )
```

2. Load a page on your site that displays a help tip. For example Products - Categories, there's a help tip to the left of the "Uncategorized" category.
3. Verify that the modifications to the help tip have been applied, inspectiing the HTML of the help tip element using your browser's inspector tool if necessary. For the above example the text of the help tip will appear in uppercase.

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [x] Have you written new tests for your changes, as applicable?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> changelog add`?
-   [x] Have you included testing instructions?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

jgreys committed:
Dev - Allow to filter wc_help_tip

jgreys committed:
Add changelog

Konamiman reviewed (changes requested):
Hi @jgreys, thanks for your contribution. I suggest a couple of changes:

1. I think it would be useful to pass the original unsanitized tip to the filter. So you would put the sanitized text into a new `$sanitized_tip` variable instead of overriding the original `$tip`, then use the sanitized version in the default HTML generated and pass both to the filter.
2. Please add a `@since 7.7.0` tag to the documentation block of the filter.

jgreys committed:
Pass sanitized/unsanitized tip to wc_help_tip filter

jgreys commented:
> Hi @jgreys, thanks for your contribution. I suggest a couple of changes:
> 
> 1. I think it would be useful to pass the original unsanitized tip to the filter. So you would put the sanitized text into a new `$sanitized_tip` variable instead of overriding the original `$tip`, then use the sanitized version in the default HTML generated and pass both to the filter.
> 2. Please add a `@since 7.7.0` tag to the documentation block of the filter.

I have updated the function to incorporate your suggestion and added a new $sanitized_tip variable to store the sanitized text. I have also updated the documentation block of the filter to include a @since 7.7.0 tag. :)

Konamiman committed:
Remove extra whitespace that makes PHPCS sad

Konamiman commented:
Thanks @jgreys, it's fine now. However I see a failing unit test that prevents the pull request from being merged. The failure doesn't seem related to your changes, could you please rebase from the latest `trunk` to see if that fixes the issue?
